.github/workflows/run-tests-on-pr.yaml

```diff
@@ -0,0 +1,13 @@
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  testing-approved-pr:
+    if: github.event.review.state == 'APPROVED'
```

Comment:
A better approach here would be that we check if the PR has a label `ready to test`. Since only maintainers can edit labels, this should be relatively secure and convenient, without messing with the actual approval of the PR.

Suggested by @jpthiele in his talk at deRSE24.